Automattic/wp-calypso
# Add WooCommerce onboarding connection track events and stop auto authorization for the new onboarding flow

This PR adds some WooCommerce Onboarding connection track events to the WooCommerce onboarding connection flow (see #34414 and p6riRB-4rG-p2 forbackground). Note that these events are only fired in this context, and not during normal Jetpack onboarding.

It also adds a few additional missing styles to the connection page, and stops auth authorization from happening for the new WooCommerce flow.

<img width="863" alt="Screen Shot 2019-07-02 at 3 46 13 PM" src="https://user-images.githubusercontent.com/689165/60542039-e7f96780-9ce0-11e9-9aad-702029274973.png">

See p1560449057014200-slack-wc-onboarding for background on the events.

Branched from https://github.com/Automattic/wp-calypso/pull/34380.

#### Testing instructions
* `npm run test-client` to make sure all tests pass.

--

If you have WooCommerce / WooCommerce Admin Installed:

* Enable the onboarding flow in `WooCommerce Admin` and to tell it to point at local Calypso. 
    * Start WooCommerce in development (not plugin) mode
    * Define the following constant in your `wp-config.php` or a plugin file: `define( 'WOOCOMMERCE_CALYPSO_LOCAL', true );`.
    * Make a `POST` request to `/wp-json/wc-admin/v1/onboarding/profile` on your local site, setting `skipped` and `completed` to false.
* Visit `/wp-admin/admin.php?page=wc-admin` on your local site, and the onboarding wizard should be displayed.
* Click `Get Started`, and then `Activate & Continue`.

If you don't want to install WooCommerce / WooCommerce Admin:

* Disconnect Jetpack from your site
* Click “Setup Jetpack”.
* On the connection/authorization screen ( https://wordpress.com/jetpack/connect/authorize?client_id= ….. ) copy and paste the whole URL.
* Replace `https://wordpress.com` with ` http://calypso.localhost:3000` and replace the `from` parameter in the URL (it may be something like `connection-banner` or `landing-page-bottom`) with  `from=woocommerce-setup-wizard` . See Pastebin `213d5`  or ping me if you have issues.

* Paste `localStorage.setItem( 'debug', 'calypso:analytics*' );` in your console.
* Test login and signup to see that events are firing.

You may run into an issue testing logged out, if your store email matches a WPCOM account. D30046-code will help with this.
